client/app/src/main/java/com/microsoft/research/karya/ui/registration/OTPFragment.kt

```diff
@@ -15,17 +19,88 @@ import com.microsoft.research.karya.ui.base.BaseActivity
 import com.microsoft.research.karya.utils.viewBinding
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import com.microsoft.research.karya.ui.registration.RegistrationViewModel.OtpVerifyState
+import com.microsoft.research.karya.ui.registration.RegistrationViewModel.OtpSendState
+import dagger.hilt.android.AndroidEntryPoint
 
 private const val OTP_LENGTH = 6
 
+@AndroidEntryPoint
 class OTPFragment : Fragment(R.layout.fragment_o_t_p) {
 
     private val binding by viewBinding(FragmentOTPBinding::bind)
+    private val viewModel by viewModels<RegistrationViewModel>()
 
     private lateinit var registrationActivity: RegistrationActivity
     private lateinit var baseActivity: BaseActivity
     private lateinit var karyaAPI: KaryaAPIService
 
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```

Comment:
Please change this to `onViewCreated`. The view inflation is done already since we have passed the layout file to the `Fragment` constructor

Reply:
Sure, I did things in onCreateview because I wanted to keep a consistent notion of attaching observers in onCreateView. This would logically seperate thing from onViewCreated since already lot of things are going on there